renderdoc/core/replay_proxy.cpp

```diff
@@ -1707,6 +1707,21 @@ bool ReplayProxy::SendReplayCommand(ReplayProxyPacket type)
   return true;
 }
 
+// Modify the params that are used when getting the texture data for replay on the current driver.
+void ReplayProxy::ConvertProxyTextureIfNeeded(FetchTexture &tex, GetTextureDataParams &params)
+{
+  if(m_Proxy->IsTextureFormatSupported(tex.format))
+    return;
+
+  tex.format.special = false;
+  tex.format.compCount = 4;
+  tex.format.compByteWidth = 1;
+  tex.format.compType = eCompType_UNorm;
+
+  params.forceRGBA8unorm = true;
+  params.whitePoint = 1.0f;
```

Comment:
Sorry, I should have been clearer before. The code was fine where it was it just needed to explicitly handle or not handle cases rather than blanket-convert to RGBA8. The D3D driver should not be responsible for choosing the remap, that should be common code in the proxy and all drivers should share it, mapping all known formats down to as small a subset as possible of formats we can just assume are always supported.

So I'd prefer to stick with the previous method of just having a `IsTextureSupported()` driver function. Then when it isn't supported, do the choosing of the remap format in the proxy as well rather than hardcoding it to RGBA8 as you had originally, or doing that in the driver as you're doing now.

There won't be a whitepoint to pick at all for HDR formats, the details will need to be decided once it's implemented but we shouldn't need to do the black/whitepoint range adaption at all for higher bit-depth formats because we should be remapping to a format with enough precision to represent all possible values. The range adaption is only needed for the original purpose of forceRGBA8unorm, when remapping a higher precision format down to be able to save it to a normal fileformat.